File: wallet/src/main/java/com/worldventures/wallet/domain/converter/WalletLocationTypeToSmartCardLocationTypeConverter.java
package com.worldventures.wallet.domain.converter;

import com.worldventures.core.converter.Converter;
import com.worldventures.dreamtrips.api.smart_card.location.model.SmartCardLocationType;
import com.worldventures.wallet.domain.entity.lostcard.WalletLocationType;

import io.techery.mappery.MapperyContext;


public class WalletLocationTypeToSmartCardLocationTypeConverter implements Converter<WalletLocationType, SmartCardLocationType> {
   @Override
   public Class<WalletLocationType> sourceClass() {
      return WalletLocationType.class;
   }

   @Override
   public Class<SmartCardLocationType> targetClass() {
      return SmartCardLocationType.class;
   }

   @Override
   public SmartCardLocationType convert(MapperyContext mapperyContext, WalletLocationType walletLocationType) {
      SmartCardLocationType smartCardLocationType = null;
      switch (walletLocationType) {
         case CONNECT:
            smartCardLocationType = SmartCardLocationType.CONNECT;
            break;
         case DISCONNECT:
            smartCardLocationType = SmartCardLocationType.DISCONNECT;
            break;
      }
      return smartCardLocationType;
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/WalletLocationTypeToSmartCardLocationTypeConverter.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.dreamtrips.api.smart_card.location.model.SmartCardLocationType
import com.worldventures.wallet.domain.entity.lostcard.WalletLocationType

import io.techery.mappery.MapperyContext

class WalletLocationTypeToSmartCardLocationTypeConverter : Converter<WalletLocationType, SmartCardLocationType> {

   override fun sourceClass(): Class<WalletLocationType> = WalletLocationType::class.java

   override fun targetClass(): Class<SmartCardLocationType> = SmartCardLocationType::class.java

   override fun convert(context: MapperyContext, source: WalletLocationType): SmartCardLocationType =
         when (source) {
            WalletLocationType.CONNECT -> SmartCardLocationType.CONNECT
            else -> SmartCardLocationType.DISCONNECT
         }
}
